packages/model-viewer/src/features/scene-graph/image.ts

```diff
@@ -41,30 +41,28 @@ export class Image extends ThreeDOMElement implements ImageInterface {
     return this[$correlatedObjects]?.values().next().value as ThreeTexture;
   }
 
-  constructor(
-      onUpdate: () => void, texture: ThreeTexture|null,
-      gltfImage: GLTFImage|null) {
-    gltfImage = gltfImage ?? {
-      name: (texture && texture.image && texture.image.src) ?
-          texture.image.src.split('/').pop() :
-          'adhoc_image',
-      uri: (texture && texture.image && texture.image.src) ?
-          texture.image.src :
-          'adhoc_image' + adhocNum++
-    };
-    super(onUpdate, gltfImage, new Set<ThreeTexture>(texture ? [texture] : []));
+  get[$threeTextures](): Set<ThreeTexture> {
+    return this[$correlatedObjects] as Set<ThreeTexture>;
+  }
+
+  constructor(onUpdate: () => void, texture: ThreeTexture|null) {
+    super(onUpdate, new Set<ThreeTexture>(texture ? [texture] : []));
+
+    if (!this[$threeTexture].image.src) {
+      this[$threeTexture].image.src = 'adhoc_image' + adhocNum++;
```

Comment:
Should this be the `name` as well, as before?

Reply:
Done